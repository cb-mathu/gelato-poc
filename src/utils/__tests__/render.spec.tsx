import { fireEvent, render, screen } from '@testing-library/vue';
import { simpleComponentFactory } from '../render';

describe('simpleComponentFactory : ', () => {
  test('renders the component with props', () => {
    const { container } = render(simpleComponentFactory('div', 'g-form'));
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the component with slots', () => {
    const GForm = simpleComponentFactory('div', 'g-form');
    const element = {
      render: () => <GForm>Hello World</GForm>,
    };
    render(element);
    expect(screen.getByText('Hello World')).toBeVisible();
  });

  test('renders the component with events attached', () => {
    const GForm = simpleComponentFactory('div', 'g-form');
    const handleClick = jest.fn();
    const element = {
      render: () => <GForm onClick={handleClick}>Hello World</GForm>,
    };
    render(element);
    fireEvent.click(screen.getByText('Hello World'));
    expect(handleClick).toHaveBeenCalled();
  });
});
