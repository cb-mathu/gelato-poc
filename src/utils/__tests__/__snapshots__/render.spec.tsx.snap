// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simpleComponentFactory :  renders the component with props 1`] = `
<div
  class="g-form"
/>
`;
